src/components/NewTaxonomy/NewTaxonomy.tsx

```diff
@@ -40,9 +40,14 @@ type TaxonomyOptions = {
   placeholder?: string,
 };
 
+type SelectedItem = {
```

Comment:
```suggestion
type SelectedItem = {
  label: string,
  value: string,
}
type SelectedItemPath = SelectedItem[]
```
Just suggesting maybe breaking this up as it was harder to understand the below logic when reading it all together as one type.